src/terminal-search-dialog.c

```diff
@@ -350,13 +353,13 @@ terminal_search_dialog_get_regex (GtkWidget *dialog)
 
 	pattern = text = terminal_search_dialog_get_search_text (dialog);
 
-	compile_flags = G_REGEX_OPTIMIZE;
+	compile_flags = PCRE2_UTF | PCRE2_NO_UTF_CHECK;
 
 	if (!GET_FLAG (match_case_checkbutton))
-		compile_flags |= G_REGEX_CASELESS;
+		compile_flags |= PCRE2_CASELESS;
 
 	if (GET_FLAG (regex_checkbutton))
-		compile_flags |= G_REGEX_MULTILINE;
+		compile_flags |= PCRE2_MULTILINE;
```

Comment:
It should be `PCRE2_UCP` instead of `PCRE2_MULTILINE`.
https://www.pcre.org/current/doc/html/pcre2posix.html